### PR TITLE
🍩 Android DonutChart interaction

### PR DESCRIPTION
**Pull request by kornelijepetak**
## What's this?
Implements interaction with the donut on the reports.

### Relationships
🎉 GO LOCO PROJECT

## Why do we want this?
So that one can see what's what among slices.

## How is it done?
By some animations.

### Why not another way?
@amulware said the other one didn't cause butterflies in the stomach.

## Review hints
Slowly and carefully.

## :squid: Permissions
Me.

### CODE REVIEW

File: Toggl.Droid/Views/DonutChart/DonutChartView.SlicesCollection.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                foreach (var percentageSegment in percentageSegments)
+                {
+                    var percentage = (float)percentageSegment.NormalizedPercentage;
+                    var sweepAngle = fullCircle * percentage;
+
+                    slices.Add(new Slice(percentageSegment, angleOffset, sweepAngle));
+                    angleOffset += sweepAngle;
+                }
+
+                slices[^1].CorrectEndAngle();
```
Comment (daividssilverio) [inline Toggl.Droid/Views/DonutChart/DonutChartView.SlicesCollection.cs]: what does the `[^1]` do? Gets the last element?

Comment (kornelijepetak) [reply Toggl.Droid/Views/DonutChart/DonutChartView.SlicesCollection.cs]: Yes.

File: Toggl.Droid/Views/DonutChart/DonutChartView.cs [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+        private void drawSelectedSliceLabels(Canvas canvas)
+        {
+            textPaint.TextSize = selectedSliceTextSize;
+
+            textPaint.Color = selectionAnimator.Color;
+            var labelPosition = center + new Point(0, selectedSliceLabelVerticalOffset);
+            drawLabel(canvas, selectedSlice.PercentageSegment.Segment.Label, labelPosition);
+
+            textPaint.Color = Color.ParseColor("#888888");
```
Comment (daividssilverio) [inline Toggl.Droid/Views/DonutChart/DonutChartView.cs]: This color shouldn't be here, right? Please add it to the `colors.xml` file (or use one from there).

**Comment by kornelijepetak** [comment]
1. Fixed.
2. Since the reports are gotten from the API, it always requires an additional request.
3. Fixed.
4. Unfortunatelly this is not that easy to solve atm. If you think it's really annoying, let's make an issue and discuss how we can solve it in future.
5. Similar to 4.

**Comment by daividssilverio** [comment]
Hey @kornelijepetak , I'm pretty sure 2. will come back as a bug report from users.
If it's not fixed here, do you mind creating an issue for it? Maybe we can fix it in regular backlog.

**Pull request merged**